torchaudio/prototype/models/_conformer_wav2vec2.py

```diff
@@ -143,6 +244,81 @@ def extract_features(
         return self._get_intermediate_outputs(x, mask=masks, num_layers=num_layers)
 
 
+class ConformerWav2Vec2PretrainModel(Module):
```

Comment:
Is it possible to define a generic `Wav2Vec2PretrainModel` so that it is compatible with original transformer-based Wav2Vec2 model. In that case, we need to keep a location for quantization model between  feature extractor and negative sampler. And I think this conformer based model still has a linear layer even though it doesn't use quantization layer.

Reply:
Just to make sure I understand correctly, to make this follow the wav2vec2 pretraining, we would change the init to take in the following args: `wav2vec2`, `mask_generator`, `linear/quanitzer` (variable name suggestion?), `ngative_sampler`? I think that's fine and generic `Wav2Vec2PretrainModel` would follow the same 4 variable usages, but should probably be out of scope of this PR and something we could add separately if we want to support. (and in that case, after adding generic `Wav2Vec2PretrainModel`, it'd be easy to subclass)

Reply:
addressed this in the latest commit, lmk what you think of this format (`intermediate` (open to other variable names) can be replaced by `quantizer` variable if adding a generic w2v2 pretrained model). another possibility would be to group the intermediate and negative sampling layers into a class, that can be fetched similarly to feature_extractor or conformer_encoder modules.

Reply:
`preprocessor` sounds good. thanks for addressing it.